hibernate-core/src/test/java/org/hibernate/test/naturalid/inheritance/cache/InheritedNaturalIdNoCacheTest.java

```diff
@@ -57,10 +58,21 @@ public void testLoadExtendedByNormal() {
 		});
 
 		doInHibernate( this::sessionFactory, session -> {
-			ExtendedEntity user = session.byNaturalId( ExtendedEntity.class )
-				.using( "uid", "base" )
-				.load();
-			assertNull( user );
+			try {
+				session.byNaturalId( ExtendedEntity.class )
+						.using( "uid", "base" )
+						.load();
+				fail( "Expecting WrongClassException" );
+			}
+			catch (WrongClassException expected) {
+				// expected outcome
+			}
+			catch (Exception other) {
+				throw new AssertionError(
+						"Unexpected exception type : " + other.getClass().getName(),
+						other
+				);
+			}
```

Comment:
This test was inconsistent with `InheritedNaturalIdCacheTest`, so I had to change one or the other of them.

We should at least define a consistent behavior for this!

Reply:
Yep, we had this same conversation in regards to 6.

The inconsistency came down whether the id-resolution was cached or not.  Personally I prefer the exception

Reply:
Fine, then my coin-toss came down on the right side.